grakn-graql/src/main/java/ai/grakn/graql/internal/query/ComputeQueryImpl.java

```diff
@@ -0,0 +1,647 @@
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static ai.grakn.util.CommonUtil.toImmutableSet;
+import static ai.grakn.util.GraqlSyntax.Char.COMMA_SPACE;
+import static ai.grakn.util.GraqlSyntax.Char.EQUAL;
+import static ai.grakn.util.GraqlSyntax.Char.QUOTE;
+import static ai.grakn.util.GraqlSyntax.Char.SEMICOLON;
+import static ai.grakn.util.GraqlSyntax.Char.SPACE;
+import static ai.grakn.util.GraqlSyntax.Char.SQUARE_CLOSE;
+import static ai.grakn.util.GraqlSyntax.Char.SQUARE_OPEN;
+import static ai.grakn.util.GraqlSyntax.Command.COMPUTE;
+import static ai.grakn.util.GraqlSyntax.Compute.ALGORITHMS_ACCEPTED;
+import static ai.grakn.util.GraqlSyntax.Compute.ALGORITHMS_DEFAULT;
+import static ai.grakn.util.GraqlSyntax.Compute.ARGUMENTS_ACCEPTED;
+import static ai.grakn.util.GraqlSyntax.Compute.Algorithm;
+import static ai.grakn.util.GraqlSyntax.Compute.Algorithm.CONNECTED_COMPONENT;
+import static ai.grakn.util.GraqlSyntax.Compute.Algorithm.K_CORE;
+import static ai.grakn.util.GraqlSyntax.Compute.Argument;
+import static ai.grakn.util.GraqlSyntax.Compute.Argument.DEFAULT_K;
+import static ai.grakn.util.GraqlSyntax.Compute.Argument.DEFAULT_MEMBERS;
+import static ai.grakn.util.GraqlSyntax.Compute.Argument.DEFAULT_MIN_K;
+import static ai.grakn.util.GraqlSyntax.Compute.CONDITIONS_ACCEPTED;
+import static ai.grakn.util.GraqlSyntax.Compute.CONDITIONS_REQUIRED;
+import static ai.grakn.util.GraqlSyntax.Compute.Condition;
+import static ai.grakn.util.GraqlSyntax.Compute.Condition.FROM;
+import static ai.grakn.util.GraqlSyntax.Compute.Condition.IN;
+import static ai.grakn.util.GraqlSyntax.Compute.Condition.OF;
+import static ai.grakn.util.GraqlSyntax.Compute.Condition.TO;
+import static ai.grakn.util.GraqlSyntax.Compute.Condition.USING;
+import static ai.grakn.util.GraqlSyntax.Compute.Condition.WHERE;
+import static ai.grakn.util.GraqlSyntax.Compute.INCLUDE_ATTRIBUTES_DEFAULT;
+import static ai.grakn.util.GraqlSyntax.Compute.Method;
+import static ai.grakn.util.GraqlSyntax.Compute.Method.CENTRALITY;
+import static ai.grakn.util.GraqlSyntax.Compute.Method.CLUSTER;
+import static ai.grakn.util.GraqlSyntax.Compute.Parameter;
+import static ai.grakn.util.GraqlSyntax.Compute.Parameter.CONTAINS;
+import static ai.grakn.util.GraqlSyntax.Compute.Parameter.K;
+import static ai.grakn.util.GraqlSyntax.Compute.Parameter.MEMBERS;
+import static ai.grakn.util.GraqlSyntax.Compute.Parameter.MIN_K;
+import static ai.grakn.util.GraqlSyntax.Compute.Parameter.SIZE;
+import static java.util.stream.Collectors.joining;
+
+
+/**
+ * Graql Compute Query: to perform distributed analytics OLAP computation on Grakn
+ *
+ * @author Haikal Pribadi
```

Comment:
add Jason

Reply:
I actually rewrote the complete implementation of the new ComputeQuery class (which didn't exist before). But we've decided to remove authors now.